frame/nfts/src/types.rs

```diff
@@ -473,3 +479,19 @@ impl CollectionRoles {
 	}
 }
 impl_codec_bitflags!(CollectionRoles, u8, CollectionRole);
+
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct PreSignedMint<CollectionId, ItemId, AccountId, Deadline> {
+	/// A collection of the item to be minted.
+	pub(super) collection: CollectionId,
+	/// Item's id.
+	pub(super) item: ItemId,
+	/// Additional item's key-value attributes.
+	pub(super) attributes: Vec<(Vec<u8>, Vec<u8>)>,
+	/// Additional item's metadata.
+	pub(super) metadata: Vec<u8>,
+	/// Restrict the claim to a particular account.
+	pub(super) only_account: Option<AccountId>,
```

Comment:
why will ever this be None? because others could frontrun the tx and steal the nft

Reply:
There are a lot of drops where the claimer is not known upfront. If that NFT won't have any monetary value, then trying to front-run the claim won't make much sense, especially when the front-runner would need to put a deposit for each NFT.